File: frontend/src/types/availability.ts
export type MatchTypePref = 'competitive' | 'relaxed' | 'any';

export interface AvailabilitySlot {
  id: string;
  user_id: string;
  date: string;
  start_time: string;
  end_time: string;
  zone_name: string | null;
  match_type_pref: MatchTypePref;
  is_active: boolean;
  created_at: string;
}

export interface AvailabilityCreate {
  date: string;
  start_time: string;
  end_time: string;
  zone_name?: string;
  latitude?: number;
  longitude?: number;
  match_type_pref: MatchTypePref;
}

File: backend/app/models/availability.py
import uuid
from datetime import date, datetime, time, timezone

from geoalchemy2 import Geography
from sqlalchemy import Boolean, Date, DateTime, ForeignKey, String, Time
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class AvailabilitySlot(Base):
    __tablename__ = "availability_slots"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id"), nullable=False
    )
    date: Mapped[date] = mapped_column(Date, nullable=False)
    start_time: Mapped[time] = mapped_column(Time, nullable=False)
    end_time: Mapped[time] = mapped_column(Time, nullable=False)
    zone_name: Mapped[str | None] = mapped_column(String(200))
    location = mapped_column(
        Geography(geometry_type="POINT", srid=4326), nullable=True
    )
    match_type_pref: Mapped[str] = mapped_column(String(20), default="any")
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

    # Relationships
    user: Mapped["User"] = relationship("User")
